File: Swift/closure/3/main.swift
//
//  main.swift
//  s
//
//  Created by Fay on 2017/5/26.
//  Copyright © 2017年 Fay. All rights reserved.
//

import Foundation

func say(_ closure: (_ string: String) -> String) {
    print(closure("Hello World!"))
}

say { (string) in
    print(string)
    return "Hello Swift!"
}

File: Lua/closure/3/main.lua
--
-- Created by IntelliJ IDEA.
-- User: fay
-- Date: 2017/5/31
-- Time: 下午3:19
-- To change this template use File | Settings | File Templates.
--

function main()
    say(lambda)
end

lambda = function(v)
    print(v)
    return "Hello Lua!"
end

function say(lambda)
    print(lambda("Hello World!"))
end

main()
